sharing/filter_range_by_datetime_vars.py

```diff
@@ -0,0 +1,234 @@
+"""
+This module filters the data by a range between two datetime values.
+"""
+
+from typing import Any, Tuple, List, Union  # pylint: disable=import-error
+from datetime import datetime, timedelta  # pylint: disable=import-error
+import re
+import pandas as pd  # pylint: disable=import-error
+from pandas.core.frame import DataFrame  # pylint: disable=import-error
+from filter_single_value import filter_single_value  # pylint: disable=import-error
+
+
+def filter_range_by_datetime_vars(
+    column: bool,
+    bracket: str,
+    lower_limit: str,
+    upper_limit: str,
+    filter_val: Union[str, int, float, None],
+    intermediate_filtered_data: DataFrame,
+    rule_is_datetime: bool,
+    rule_is_numeric: bool,
+    rule_is_char: bool,
+    datetime_variables: List[Any],
+    numeric_variables: List[Any],
+    string_variables: List[Any],
+) -> Tuple[DataFrame, bool, bool, bool]:
+    """The function filters data by the datetime variables in current rule.
+
+    Function checks if the current rule value is of datetime type using
+    regular expression. If it is, then filters data by datetime variables
+    in current rule.
+    Function returns back the dataframe with or without filter operation.
+
+    Parameter:
+        column(bool): checks whether the rule direction is column
+        bracket(str): type of range bracket
+        lower_limit(str): lower limit of the range
+        upper_limit(str): upper limit of the range
+        filter_val(str): single value to filter
+        intermediate_filtered_data(dataframe): data to be filtered
+        rule_is_datetime(bool): bool variable checks rule value is datetime type
+        rule_is_numeric(bool): bool variable checks rule value is numeric type
+        rule_is_char(bool): bool variable checks rule value is string type
+        datetime_variables(list): current rule variables of datetime type
+        numeric_variables(list): current rule variables of numeric type
+        string_variables(list): current rule variables of string type
+    Returns:
+        dataframe: Dataframe that is filtered if rule value is of datetime type
+        bool: bool variable specifies the data type of rule
+    """
+
+    # Define pattern for the datetime type
+    pat1: str = r"(\d+/\d+/\d+)"
```

Comment:
Rename all these patterns to be more meaningful. They do not say what kind of date time pattern they're trying to represent.